Contact Update Report/Update and Diff activity

jaapjansma opened the pull request:
When one sets a override field and/or override details. Then this data is not added to the diff activity.

Expected behaviour
The diff activity is created listing the old data and the submitted data
The submitted data is stored at the contact, overriding existing data.

Current behaviour
The data is overridden at the contact record but not listed in the diff activity.

jensschuppe commented:
@bjendres Isn't that expected behavior, since the diff activity should only include changes that haven't been applied instantly by XCM?

Quoting from the [docs](https://docs.civicrm.org/xcm/en/latest/configuration/#difference-handling):
> This section defines, what to do when a contact has been identified by XCM, submitted data was different to the contact's current state, and those different properties have not been changed by XCM.

jaapjansma commented:
Ah I could not find that. I did not know it was intended. However I have two clients who wants to see the applied changes. How should we proceed?

bjendres commented:
Hm, the diff activity is explicitly designed to be an overview of the remaining differences. What you want, @jaapjansma, is some sort of a update report - correct? This should be a completely separate feature then, don't you agree?

jaapjansma commented:
Yeah what you call an update report is what I need.

jaapjansma commented:
What about this?

![Screenshot_20200619_154706](https://user-images.githubusercontent.com/4126292/85139087-341c1f80-b244-11ea-8e02-1170b78a033f.png)

And the help text:

![Screenshot_20200619_154716](https://user-images.githubusercontent.com/4126292/85139084-33838900-b244-11ea-8b28-508c23894a03.png)

bjendres commented:
@jaapjansma could we move this discussion to #64?

bjendres commented:
(and: I would personally clone the "Difference Handling" section to a "Updates Performed" section)

jaapjansma commented:
I have updated the PR with an option to set the status and renamed the drop down options.
![Screenshot_20200619_163008](https://user-images.githubusercontent.com/4126292/85143605-400ae000-b24a-11ea-8a94-a1d294bdf841.png)

jaapjansma commented:
I have renamed the handlers to:

* "Contact Value Comparison" (for all data, i.e. Jaap's new thing)
* "Unresolved Differences" (for the old feature)
